Update the README file to reflect the decisions made during the team meetup

bobsilverberg opened the pull request:
Also add a LICENSE file

This commit should include all of the changes documented in the etherpad [1] in the section for "9/15/15 10:00am to 12:00pm Project READMEs". Specifically, I attempted to address everything under the bullet point entitled "Changes decided - using https://github.com/mozilla/Addon-Tests/blob/master/README.md as an example".

As this was a team effort, and we are hoping to use this as a template for all of our other repos, I would appreciate a review from all team members. That includes @davehunt, @m8ttyB, @rbillings, @krupa and @stephendonner.

Thanks everyone!

rbillings commented on README.md:
```diff
@@ -1,162 +1,74 @@
-[webqa]: http://quality.mozilla.org/teams/web-qa/
-[firefox]: http://quality.mozilla.org/teams/desktop-firefox/
 [webdriver]: http://seleniumhq.org/docs/03_webdriver.html
+[runningtests]: https://developer.mozilla.org/en-US/docs/Mozilla/QA/Running_Web_QA_automated_tests
```
I think it would be better to move the links for webdriver & runningtests above the contributors blob to line 18, or below into the questions area. They are free floating right now which removes context.

rbillings commented:
This looks a lot cleaner and easier to read. My only suggestion was added. Thanks @bobsilverberg

davehunt commented on README.md:
```diff
@@ -1,162 +1,74 @@
+[![stage](https://img.shields.io/jenkins/s/https/webqa-ci.mozilla.com/amo.stage.svg?label=stage)](https://webqa-ci.mozilla.com/view/AMO/job/amo.stage/)
+[![prod](https://img.shields.io/jenkins/s/https/webqa-ci.mozilla.com/amo.prod.svg?label=prod)](https://webqa-ci.mozilla.com/view/AMO/job/amo.prod/)
 
 Getting involved as a contributor
```
I think 'as a contributor' is implied, and don't see that it's needed in the heading.

davehunt commented on README.md:
```diff
@@ -1,162 +1,74 @@
 
-[mozwebqa]:http://02.chat.mibbit.com/?server=irc.mozilla.org&channel=#mozwebqa
-[mozillians]:http://02.chat.mibbit.com/?server=irc.mozilla.org&channel=#mozillians
+[mozwebqa]:http://widget01.mibbit.com/?settings=1b10107157e79b08f2bf99a11f521973&server=irc.mozilla.org&channel=%23mozwebqa
```
This URL does look fragile, is there anything we can do about that? Perhaps we could direct people to a single place that contains all the ways they can get in touch? Note this is fine as is, I'm just thinking of the next round of improvements we could make. :)

davehunt commented on README.md:
```diff
@@ -1,162 +1,74 @@
-[GitWin]: http://help.github.com/win-set-up-git/
-[GitMacOSX]: http://help.github.com/mac-set-up-git/
-[GitLinux]: http://help.github.com/linux-set-up-git/
+###Clone the repo
```
...sitory (personal preference, I don't like using 'repo' in documentation)

davehunt commented on README.md:
```diff
@@ -1,162 +1,74 @@
-[GitMacOSX]: http://help.github.com/mac-set-up-git/
-[GitLinux]: http://help.github.com/linux-set-up-git/
+###Clone the repo
+If you have cloned this project already then you can skip this, otherwise you'll need to clone this repo using Git. If you do not know how to clone a GitHub repository, check out this [help page] (https://help.github.com/articles/cloning-a-repository/) from GitHub.
```
Should we mention forking the repository?

davehunt replied:
Also, general comment, can we keep the lines < 100 characters so they're easy to see in an editor without word wrap?

bobsilverberg replied:
@davehunt We discussed the question of forking in the meeting and my recollection was that we decided that is something we could discuss with them if they got to the point of wanting to help maintain the tests. That is why I simply left it out.

davehunt replied:
The problem there is that they then need to change their origin to be their fork and to add an upstream. We either have the upfront cost of clicking a button or the later cost of some command line changes. I'm 50/50 but we should at least have a decent resource ready for the case of a subsequent fork.

davehunt commented on README.md:
```diff
@@ -1,162 +1,74 @@
-[Python]: http://www.python.org/download/releases/2.6.6/
+###Create or activate a Python virtual environment
+You should install this project's dependencies (which is described in the next step) into a virtual environment in order to avoid impacting the rest of your system, and to make problem solving easier. If you already have a virtual environment for these tests, then you should activate it, otherwise you should create a new one.
+For more information on working with virtual environments see our [our quickstart guide] (https://wiki.mozilla.org/QA/Execution/Web_Testing/Automation/Virtual_Environments) and also [this blog post] (http://www.silverwareconsulting.com/index.cfm/2012/7/24/Getting-Started-with-virtualenv-and-virtualenvwrapper-in-Python).
```
I like this, thanks!

davehunt commented on README.md:
```diff
@@ -1,162 +1,74 @@
-[AddonsDev]: https://addons-dev.allizom.org/en-US/firefox/
+### Create a variables.json file
+Some of the tests in this repo need to log into the addons website and/or PayPal. We store the credentials (i.e., username and password) to access those sites in a file called `variables.json`, which we then pass to the tests via the command line. If you want to be able to run any of those tests, you will need your own copy of the `variables.json` file, which you will update to contain your own c ...
+To do that, make a copy of the `variables.json` file which exists in the project root and update that with your own credentials. You will then pass the name of that file on the command line. For the purposes of the examples below, assume you named your copy of the file `my_variables.json`.
```
Is this enough information? It doesn't cover how to determine the values for the file, which I suspect is one of the main barriers.

bobsilverberg replied:
This is a good point, and I'm not even sure what to tell them. Does anyone want to take a stab at documenting that?

davehunt replied:
Yeah, let's make it a separate improvement. I even think some of the values for this repository are now unused, which would make it simpler. Could you raise an issue for detailing the steps to populate a variables file for this repository?

davehunt commented on README.md:
```diff
@@ -1,162 +1,74 @@
+To run all of the desktop tests against the default environment, which is the addons development environment:
 
-	py.test --driver=firefox --variables=/path/to/variables.json tests/desktop/test_details_page.py
+	py.test --driver=firefox --variables=/path/to/my_variables.json tests/desktop
```
we could drop the `/path/to/` for brevity

davehunt commented on README.md:
```diff
@@ -1,162 +1,74 @@
+To run against a different environment, pass in a value for --baseurl, like so:
 
-	py.test --driver=firefox --variables=/path/to/variables.json tests/desktop/test_details_page.py -k test_that_external_link_leads_to_addon_website
+	py.test --driver=firefox --variables=/path/to/my_variables.json --baseurl=https://addons.allizom.org tests/desktop
```
I would suggest putting `--baseurl` as the first argument for clarity.

davehunt commented on README.md:
```diff
@@ -1,162 +1,74 @@
-[GitHub Templates]: https://github.com/mozilla/mozwebqa-test-templates 
-[Style Guide]: https://wiki.mozilla.org/QA/Execution/Web_Testing/Docs/Automation/StyleGuide
-
 License
```
We talked about dropping this now we have the badge.

davehunt commented on README.md:
```diff
@@ -1,162 +1,74 @@
+[![travis](https://img.shields.io/travis/mozilla/Addon-Tests.svg?label=travis)](http://travis-ci.org/mozilla/Addon-Tests/)
+[![dev](https://img.shields.io/jenkins/s/https/webqa-ci.mozilla.com/amo.dev.svg?label=dev)](https://webqa-ci.mozilla.com/view/AMO/job/amo.dev/)
+[![stage](https://img.shields.io/jenkins/s/https/webqa-ci.mozilla.com/amo.stage.svg?label=stage)](https://webqa-ci.mozilla.com/view/AMO/job/amo.stage/)
+[![prod](https://img.shields.io/jenkins/s/https/webqa-ci.mozilla.com/amo.prod.svg?label=prod)](https://webqa-ci.mozilla.com/view/AMO/job/amo.prod/)
```
I love *love* ***love*** the badges!

davehunt commented:
This is great, thanks @bobsilverberg. I added a few comments, but r+ with those addressed.

bobsilverberg commented:
Thanks for the reviews @rbillings and @davehunt. I have addressed most of your comments, and the ones I have not I added a response. Unfortunately GitHub thinks it's smarter than us and has removed all those comments from the conversation flow because I changed the related lines in the file, so you may have to check email for them. 

I'd be happy to discuss them further before considering merging this.

davehunt commented on README.md:
```diff
@@ -1,168 +1,96 @@
-----------------------
+[![license](https://img.shields.io/badge/license-MPL%202.0-blue.svg)](https://github.com/mozilla/Addon-Tests/blob/master/LICENSE)
+[![travis](https://img.shields.io/travis/mozilla/Addon-Tests.svg?label=travis)](http://travis-ci.org/mozilla/Addon-Tests/)
+[![dev](https://img.shields.io/jenkins/s/https/webqa-ci.mozilla.com/amo.dev.svg?label=dev)](https://webqa-ci.mozilla.com/view/AMO/job/amo.dev/)
```
Just a thought, but I think we should drop the view from the URLs in case we rename or remove it in the future. It's not likely, but we can instead point directly at the job.

davehunt commented on README.md:
```diff
@@ -1,168 +1,96 @@
+[![license](https://img.shields.io/badge/license-MPL%202.0-blue.svg)](https://github.com/mozilla/Addon-Tests/blob/master/LICENSE)
+[![travis](https://img.shields.io/travis/mozilla/Addon-Tests.svg?label=travis)](http://travis-ci.org/mozilla/Addon-Tests/)
+[![dev](https://img.shields.io/jenkins/s/https/webqa-ci.mozilla.com/amo.dev.svg?label=dev)](https://webqa-ci.mozilla.com/view/AMO/job/amo.dev/)
+[![stage](https://img.shields.io/jenkins/s/https/webqa-ci.mozilla.com/amo.stage.svg?label=stage)](https://webqa-ci.mozilla.com/view/AMO/job/amo.stage/)
```
It appears this job does not exist.

bobsilverberg commented:
@davehunt I fixed/updated the badge links, and added a paragraph about forking the repo(sitory). Please take a look and let me know what you think.

davehunt commented:
r+
